circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/CopierFactoryManager.java

```diff
@@ -55,14 +57,26 @@ void postConstruct() {
     }
   }
 
-  CopierFactory getCopierFactory(Path sourceLocation, Path replicaLocation) {
+  CopierFactory getCopierFactory(Path sourceLocation, Path replicaLocation, Map<String, Object> copierOptions) {
     String sourceScheme = sourceLocation.toUri().getScheme();
     String replicaScheme = replicaLocation.toUri().getScheme();
-    for (CopierFactory copierFactory : copierFactories) {
-      if (copierFactory.supportsSchemes(sourceScheme, replicaScheme)) {
-        LOG.debug("Found CopierFactory '{}' for sourceScheme '{}' and replicaScheme '{}'",
-            copierFactory.getClass().getName(), sourceScheme, replicaScheme);
-        return copierFactory;
+    if (copierOptions.containsKey(COPIER_FACTORY_CLASS)) {
+      for (CopierFactory copierFactory : copierFactories) {
+        final String copierFactoryClassName = copierFactory.getClass().getName();
+        if (copierFactoryClassName.equalsIgnoreCase(copierOptions.get(COPIER_FACTORY_CLASS).toString())) {
```

Comment:
Minor, but why is this check case insensitive?

Reply:
Just a safety check, if the yaml is not in the correct case

Reply:
Yeah, I think I'd prefer this to be case sensitive and leave the onus on the config file to do things properly.